apps/opik-backend/src/main/java/com/comet/opik/infrastructure/redis/RedisUrl.java

```diff
@@ -0,0 +1,34 @@
+package com.comet.opik.infrastructure.redis;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public record RedisUrl(String scheme, String host, int port, int database) {
+
+    public static RedisUrl parse(String redisUrl) {
+        try {
+            URI uri = new URI(redisUrl);
+            String scheme = uri.getScheme();
+            String host = uri.getHost();
+            int port = uri.getPort() == -1 ? 6379 : uri.getPort();
+            String path = uri.getPath();
+            int database = 0;
+
+            if (path != null && path.length() > 1) {
```

Comment:
Can't the database index be one digit?

Reply:
Yes, if you check the `getDatabase` it does substring from index 1

Reply:
Great, I suspected that when I noticed the tests cover this scenario

Reply:
Yeah, I also just added the credentials scenario